cal/cal/static/js/app.js

```diff
@@ -3,109 +3,171 @@ var analyticsApp = angular.module('analyticsApp', ['nvd3', 'ui.calendar']);
-          tag.label = data.label;
-          tag.keywords = data.keywords;
-          tag.hours = data.hours;
-        });
-    };
-
-    this.cancelEdit = function(tagId) {
-      var tag = $scope.tags.find(function(tag, index, array) { return tag.id == tagId; });
-      tag.editing = false;
-    };
-
-    this.delete = function(tagId) {
-      $http({
-        method: 'POST',
-        url: tagUrl + '/' + tagId,
-        data: $.param({
-          csrfmiddlewaretoken: getCookie('csrftoken'),
-          _method: 'DELETE'
-        }),
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded'
         }
-      }).
-        success(function removeFromList(data) {
-          $scope.tags = $scope.tags.filter(function(tag) {
-            return tag.id !== tagId;
-          });
-        });
-    };
+      });
+  }
+  var initialTimeRange = CalendarRangeService.getRange()
+  this.updateTags(initialTimeRange.timeRange, initialTimeRange.start, initialTimeRange.end);
+
+  this.hideZeroHoursFilter = function (value, index, array) {
+    if (this.hideZeroHours && value.hours === 0) {
+      return false;
+    } else {
+      return true;
+    }
+  }.bind(this);
+
+  this.create = function(tag) {
+    $http({
+      method: 'POST',
+      url: tagUrl + '.json',
+      data: $.param({
+        label: tag.label,
+        keywords: tag.keywords,
+        csrfmiddlewaretoken: getCookie('csrftoken')
+      }),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }).
+    success(function addToList(data) {
+      this.tags[this.timeRange].push({
+        id: data.id,
+        label: data.label,
+        keywords: data.keywords,
+        hours: data.hours,
+        editing: false
+      });
+    });
+  }.bind(this);
+
+  this.startEdit = function(tagId) {
+    var tag = this.tags[this.timeRange].find(function(tag, index, array) { return tag.id == tagId; });
+    tag.newLabel = tag.label;
+    tag.newKeywords = tag.keywords;
+    tag.editing = true;
+  };
+
+  this.submit = function(tagId) {
+    var tag = this.tags[this.timeRange].find(function(tag, index, array) { return tag.id == tagId; });
+    tag.editing = false;
+
+    var addToList = function addToList(data) {
+      tag.label = data.label;
+      tag.keywords = data.keywords;
+      tag.hours = data.hours;
+    }.bind(this);
+
+    $http({
+      method: 'POST',
+      url: tagUrl + '/' + tagId,
+      data: $.param({
+        label: tag.newLabel,
+        keywords: tag.newKeywords,
+        csrfmiddlewaretoken: getCookie('csrftoken'),
+        _method: 'PATCH'
+      }),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }).success(addToList);
+  }.bind(this);
+
+  this.cancelEdit = function(tagId) {
+    var tag = this.tags[this.timeRange].find(function(tag, index, array) { return tag.id == tagId; });
+    tag.editing = false;
+  }.bind(this);
+
+  var removeFromList = function removeFromList(data) {
```

Comment:
Removed this variable assignment so that there isn't confusion on our controller api.